toolchain/check/testdata/where_expr/constraints.carbon

```diff
@@ -19,23 +19,101 @@ interface I {
   let Second:! J;
 }
 
-fn Equal(T:! I where .Member = {});
-
 fn EqualEqual(U:! I where .Self == ());
 
 fn Impls(V:! J where .Self impls I);
 
-fn And(W:! I where .Second impls I and .Member = .Second);
+fn And(W:! I where .Self impls J and .Member == ());
+
+// --- fail_todo_equal_constraint.carbon
+
+library "[[@TEST_NAME]]";
+
+interface N {
+  let P:! type;
+}
+
+// CHECK:STDERR: fail_todo_equal_constraint.carbon:[[@LINE+7]]:27: error: cannot implicitly convert from `{}` to `<associated type in N>`
```

Comment:
I assume this is intended to eventually compile without errors. Would it be straightforward to get a TODO diagnostic here in the meantime? It's not a big deal, but if someone stumbles across this error during development, letting them know this is just a missing feature could save them a wild goose chase.

Reply:
Unfortunately, it is not straightforward for me. There is a bug in the implementation of member access for facet types. @zygoloid and I started looking at how to fix it, but it turned out to be a bigger project. I think we are going to need to do some more foundational work in the representation of facet values and facet types first. I am not aware of an easy way to detect this situation at the moment. You can see how far we got in https://github.com/carbon-language/carbon-lang/pull/4371.